process/block/sovereign/outgoingOperations_test.go

```diff
@@ -25,115 +25,376 @@ func createEvents() []SubscribedEvent {
+func createValidArgs() *outgoingOperations {
+	dataCodec := &mock.DataCodecMock{}
+	topicsChecker := &mock.TopicsCheckerMock{
+		CheckValidityCalled: func(topics [][]byte) error {
+			return nil
+		},
+	}
+
+	events := []SubscribedEvent{
+		{
+			Identifier: []byte("deposit"),
+			Addresses: map[string]string{
+				"addr1": "addr1",
+				"addr2": "addr2",
+			},
+		},
+	}
+
+	args := ArgsOutgoingOperations{
+		SubscribedEvents: events,
+		DataCodec:        dataCodec,
+		TopicsChecker:    topicsChecker,
+	}
+	creator, _ := NewOutgoingOperationsFormatter(args)
+	return creator
+}
+
+func TestOutgoingOperations_CheckEvent(t *testing.T) {
 	t.Parallel()
 
-	addr1 := []byte("addr1")
-	addr2 := []byte("addr2")
-	addr3 := []byte("addr3")
+	t.Run("invalid identifier", func(t *testing.T) {
+		t.Parallel()
 
-	identifier1 := []byte("deposit")
-	identifier2 := []byte("send")
+		events := []SubscribedEvent{
+			{
+				Identifier: []byte(""),
+				Addresses: map[string]string{
+					"addr1": "addr1",
+				},
+			},
+		}
+
+		args := ArgsOutgoingOperations{
+			SubscribedEvents: events,
+			DataCodec:        &mock.DataCodecMock{},
+			TopicsChecker:    &mock.TopicsCheckerMock{},
+		}
+		creator, err := NewOutgoingOperationsFormatter(args)
+		require.Nil(t, creator)
+		require.ErrorContains(t, err, "no subscribed identifier")
+	})
+	t.Run("no addresses", func(t *testing.T) {
+		t.Parallel()
+
+		events := []SubscribedEvent{
+			{
+				Identifier: []byte("identifier"),
+				Addresses:  map[string]string{},
+			},
+		}
+
+		args := ArgsOutgoingOperations{
+			SubscribedEvents: events,
+			DataCodec:        &mock.DataCodecMock{},
+			TopicsChecker:    &mock.TopicsCheckerMock{},
+		}
+		creator, err := NewOutgoingOperationsFormatter(args)
+		require.Nil(t, creator)
+		require.ErrorContains(t, err, errNoSubscribedAddresses.Error())
+	})
+	t.Run("invalid address", func(t *testing.T) {
+		t.Parallel()
+
+		events := []SubscribedEvent{
+			{
+				Identifier: []byte("identifier"),
+				Addresses: map[string]string{
+					"addr1": "",
+				},
+			},
+		}
+
+		args := ArgsOutgoingOperations{
+			SubscribedEvents: events,
+			DataCodec:        &mock.DataCodecMock{},
+			TopicsChecker:    &mock.TopicsCheckerMock{},
+		}
+		creator, err := NewOutgoingOperationsFormatter(args)
+		require.Nil(t, creator)
+		require.ErrorContains(t, err, errNoSubscribedAddresses.Error())
+	})
+}
+
+func TestOutgoingOperations_OutgoingEvents(t *testing.T) {
+	t.Parallel()
+
+	logs := []*data.LogData{
+		{
+			LogHandler: &transactionData.Log{
+				Address: nil,
+				Events: []*transactionData.Event{
+					{
+						Address:    []byte("addr1"),
+						Identifier: []byte("deposit"),
+						Topics:     [][]byte{[]byte("topic1"), []byte("topic1"), []byte("topic1"), []byte("topic1"), []byte("topic1")},
+						Data:       []byte("data"),
+					},
+				},
+			},
+			TxHash: "",
+		},
+	}
+
+	t.Run("nil logs", func(t *testing.T) {
+		t.Parallel()
+
+		outgoingOps := createValidArgs()
```

Comment:
You could only use one instance, not multiple

Reply:
not really

Reply:
Here in these tests you should've actually rewrite args with mocks, not the object's internal struct